src/app_service/service/chat/async_tasks.nim

```diff
@@ -27,7 +27,11 @@ type
 const asyncCheckChannelPermissionsTask: Task = proc(argEncoded: string) {.gcsafe, nimcall.} =
   let arg = decode[AsyncCheckChannelPermissionsTaskArg](argEncoded)
   try:
-    let response = status_communities.checkCommunityChannelPermissions(arg.communityId, arg.chatId)
+    var response = status_communities.checkCommunityChannelPermissionsLight(arg.communityId, arg.chatId).result
+    let channelPermissions = response.toCheckChannelPermissionsResponseDto()
+    if  not channelPermissions.viewOnlyPermissions.satisfied and not channelPermissions.viewAndPostPermissions.satisfied:
+      response = status_communities.checkCommunityChannelPermissions(arg.communityId, arg.chatId).result
+
     arg.finish(%* {
       "response": response,
```

Comment:
in my taste it is better to separate result and error, as it is done in other async rpc